Auth0/JWKS.swift

```diff
@@ -0,0 +1,91 @@
+// JWKS.swift
+//
+// Copyright (c) 2019 Auth0 (http://auth0.com)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import Foundation
+import SimpleKeychain
+
+public struct JWKS: Codable {
+    let keys: [JWK]
+}
+
+public extension JWKS {
+    func key(id kid: String) -> JWK? {
+        return keys.first { $0.keyId == kid }
+    }
+}
+
+public struct JWK: Codable {
+    let keyType: String
+    let keyId: String?
+    let usage: String?
+    let algorithm: String?
+    let certUrl: String?
+    let certThumbprint: String?
+    let certChain: [String]?
+    let rsaModulus: String?
+    let rsaExponent: String?
+
+    enum CodingKeys: String, CodingKey {
+        case keyType = "kty"
+        case keyId = "kid"
+        case usage = "use"
+        case algorithm = "alg"
+        case certUrl = "x5u"
+        case certThumbprint = "x5t"
+        case certChain = "x5c"
+        case rsaModulus = "n"
+        case rsaExponent = "e"
+    }
+}
+
+extension JWK {
+    var rsaPublicKey: SecKey? {
+        guard let modulus = rsaModulus?.a0_decodeBase64URLSafe(),
+            let exponent = rsaExponent?.a0_decodeBase64URLSafe() else { return nil }
+        let encodedKey = encodeRSAPublicKey(modulus: [UInt8](modulus), exponent: [UInt8](exponent))
+        if #available(iOS 10, OSX 10.12, tvOS 10, watchOS 3, *) {
+            return generateRSAPublicKey(from: encodedKey)
+        }
+        let tag = "com.auth0.tmp.RSAPublicKey"
+        let keychain = A0SimpleKeychain()
+        guard keychain.setRSAPublicKey(data: encodedKey, forKey: tag) else { return nil }
+        return keychain.keyRefOfRSAKey(withTag: tag).takeRetainedValue()
```

Comment:
This fails on macOS 10.11 (El Capitan), which happens to be the minimum supported version. So in that specific macOS version, the signature validation is skipped. Check https://github.com/auth0/Auth0.swift/pull/333/files#diff-2345af69c12dd8c93d94fdb71f422ba3R38.

In https://github.com/auth0/Auth0.swift/pull/333/files#diff-622b00adee0ac311c480485596339902R38 and https://github.com/auth0/Auth0.swift/pull/333/files#diff-b2bf319ef5a2f202615bba8e8cd28887R50, version checks were added to acomodate this.